ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/TransactionService.scala

```diff
@@ -898,11 +898,9 @@ class TransactionService(session: LedgerSession) extends LedgerTestSuite(session
     case Participants(Participant(alpha, operator, receiver), Participant(beta, giver)) =>
       for {
         agreementFactory <- beta.create(giver, AgreementFactory(receiver, giver))
-        agreement <- alpha
-          .exerciseAndGetContract[Agreement](
-            receiver,
-            agreementFactory.exerciseAgreementFactoryAccept,
-          )
+        agreement <- eventually {
```

Comment:
Perhaps use `beta.waitForParties(Seq(alpha), Set(giver, receiver))` before this instead?

Reply:
I think we mostly went with wrapping in `eventually`.

Reply:
If I'm not mistaken `waitForParties` is already part of the test framework allocating the parties for the test. The error is raised when the test is ran with the flag to not wait for parties against a ledger that doesn't expose parties to remote participants.

Reply:
Ooooh, now I understand. Apologies.